ydb/src/client_topic/topicwriter/writer.rs

```diff
@@ -0,0 +1,405 @@
+                ServerMessage::InitResponse(init_response_body) => {
+                    return Err(YdbError::Custom(
+                        "Unexpected message type in stream reader: init_response".to_string(),
+                    ));
+                }
+                ServerMessage::WriteResponse(write_response_body) => {
+                    for raw_ack in write_response_body.acks {
+                        let write_ack = WriteAck::from(raw_ack);
+                        let mut reception_queue = confirmation_reception_queue.lock().unwrap();
+                        let reception_ticket = reception_queue.try_get_ticket();
+                        match reception_ticket {
+                            None => {
+                                return Err(YdbError::Custom(
+                                    "Expected reception ticket to be actually present".to_string(),
+                                ));
+                            }
+                            Some(mut ticket) => {
+                                if write_ack.seq_no != ticket.get_seq_no() {
+                                    return Err(YdbError::Custom(
+                                        "Reception ticket and write ack seq_no mismatch"
+                                            .to_string(),
+                                    ));
+                                }
+                                ticket.send_confirmation_if_needed(write_ack.status);
+                            }
+                        }
+                    }
+                }
+                ServerMessage::UpdateTokenResponse(_update_token_response_body) => {}
+            },
+            Err(some_err) => {
+                return Err(YdbError::from(some_err));
+            }
+        }
+        Ok(())
+    }
+
+    pub async fn write(&mut self, mut message: TopicWriterMessage) -> YdbResult<()> {
+        self.is_cancelled().await?;
+
+        self.last_seq_num_handled += 1;
+        let message_seq_num = self.last_seq_num_handled;
+        let mut reception_queue = self.confirmation_reception_queue.lock().unwrap();
+
+        if self.auto_set_seq_no {
+            message.seq_no = message_seq_num;
+        }
+
+        reception_queue.add_ticket(TopicWriterReceptionTicket::new(
+            message_seq_num,
+            TopicWriterReceptionType::NoConfirmationExpected,
+        ));
+        self.writer_message_sender.borrow_mut().send(message).await; // TODO: handle
+
+        println!("message is sent");
+        Ok(())
+    }
+
+    pub async fn stop(self) -> YdbResult<()> {
+        println!("Stopping...");
+
+        self.flush().await?;
+        self.cancellation_token.cancel();
+
+        self.writer_loop.await; // TODO: handle ERROR
+        println!("Writer loop stopped");
+
+        self.receive_messages_loop.await; // TODO: handle ERROR
+        println!("Message receive stopped");
+        Ok(())
+    }
+
+    pub async fn write_with_ack(
```

Comment:
The write_with_ack function will not send message to a server :(

I suggest to use one function for send message, then call them from both: write and write_with_ack, for prevent different logic.